services/rewarding/open_category_reward.py

```diff
@@ -0,0 +1,275 @@
+        dependencies: dict[list],
+        images: list,
+        mode="hybrid",
+    ):
+        """Get reward for the generated questions use structured question graph.
+        Args:
+            questions (list[str]): a list of questions generated based on the tuples
+            dependencies (dict[list]): the dependencies between tuples
+            images (list[str]): a list of image urls
+        """
+
+        self.device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.binary_vqa.to(self.device)
+        scores = {}
+
+        sorted_questions, dependencies = self._create_graph_questions(questions, dependencies)
+        print(sorted_questions)
+
+        for i in range(len(images)):
+            scores[i] = [0] * len(sorted_questions)
+
+        def get_reward_for_a_question(
+            question: str,
+            question_dependencies: list[int],
+            image: Image.Image,
+            prev_scores: list[int],
+        ) -> float:
+            if any([not (prev_scores[i] > 0.5) for i in question_dependencies]):
+                print(
+                    f"Skipping question: {question}. It depends on {[sorted_questions[i] for i in range(len(question_dependencies))]} that was answered as No."
+                )
+                return 0
+            if not isinstance(image, Image.Image):
+                raise ValueError("Invalid image type")
+
+            inputs = self.binary_vqa_processor(text=question, images=image, return_tensors="pt").to(self.device, torch.float16)
+            decoder_input_ids = torch.LongTensor([[self.binary_vqa.language_model.config.pad_token_id, self.binary_vqa.language_model.config.decoder_start_token_id]]).to(self.device)
+            outputs = self.binary_vqa(
+                input_ids=inputs["input_ids"],
+                pixel_values=inputs["pixel_values"],
+                decoder_input_ids=decoder_input_ids
+            )
+            logits = outputs.logits[:, -1]
+            score = logits[0].sigmoid().item()
+            print(question)
+            print(f"The answer Yes has {score} probs")
+            return score
+
+        pbar = tqdm(
+            total=len(sorted_questions) * len(images),
+            desc=f"Calculating reward over {len(images)} images and {len(sorted_questions)} questions",
+        )
+        for i, question in enumerate(sorted_questions):
+            for j, image in enumerate(images):
+                scores[j][i] = get_reward_for_a_question(
+                    question, dependencies[i], image, scores[j]
+                )
+                pbar.update(1)
+
+        return scores, sorted_questions
+
+class IQA:
+    def __init__(self, model_name="nima-vgg16-ava"):
+        import pyiqa
+        device = torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+        self.model = pyiqa.create_metric(model_name, device=device)
+    def __call__(self, image_path):
+        return self.model(image_path)
```

Comment:
i'm thinking of miner should return PNG formatted image, currently is JPG for fixed category. And we should verify it by pillow function
https://stackoverflow.com/questions/32679589/how-to-get-the-format-of-image-with-pil

About the resolution, we should assert miner return size in one of:
```python
{
    '1:1': [
        1024,
        1024
    ],
    '3:2': [
        1152,
        768
    ],
    '2:3': [
        768,
        1152
    ],
    '19:13': [
        1216,
        832
    ],
    '13:19': [
        832,
        1216
    ],
    '16:9': [
        1360,
        768
    ],
    '9:16': [
        768,
        1360
    ]
}
```